plugins/woocommerce/src/Blocks/BlockTypes/ProductImageGallery.php

```diff
@@ -67,7 +69,7 @@ protected function render( $attributes, $content, $block ) {
 		$product_image_gallery_html = ob_get_clean();
 
 		$product   = $previous_product;
-		$classname = $attributes['className'] ?? '';
+		$classname = StyleAttributesUtils::get_classes_by_attributes( $attributes );
```

Comment:
For safety reason, I would pass the second parameter: 
```suggestion
		$classname                      = StyleAttributesUtils::get_classes_by_attributes( $attributes, array('extra_classes') );
```

Furthermore, `get_classes_by_attributes` returns an object with `classes` and `styles` keys. We need to pick the `class` property.

Reply:
Same as https://github.com/woocommerce/woocommerce/pull/50662#discussion_r1775980736